components-java-sdk/src/test/java/io/dapr/components/wrappers/InputBindingGrpcComponentWrapperTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.dapr.components.TestUtils.assertThrowsGrpcRuntimeException;
+import static io.dapr.components.TestUtils.monoError;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class InputBindingGrpcComponentWrapperTest {
+
+  /**
+   * This rule manages automatic graceful shutdown for the registered servers and channels at the
+   * end of test.
+   */
+  @Rule
+  public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  private InputBinding mockComponent;
+  private InputBindingGrpc.InputBindingBlockingStub client;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    mockComponent = mock();
+
+    // Generate a unique in-process server name.
+    final String serverName = InProcessServerBuilder.generateName();
+
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    grpcCleanup.register(InProcessServerBuilder
+        .forName(serverName)
+        .directExecutor()
+        .addService(new InputBindingGrpcComponentWrapper(mockComponent))
+        .build()
+        .start()
+    );
+
+    client = InputBindingGrpc.newBlockingStub(
+        // Create a client channel and register for automatic graceful shutdown.
+        grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build()));
+  }
+
+
+  //
+  // Tests for common component aspects
+  //
+
+  @Test
+  void initHappyCase() {
+    final Map<String, String> expectedMetadata = Map.of("key", "value");
+
+    when(mockComponent.init(anyMap())).thenReturn(Mono.empty());
+
+    final var response = client.init(Bindings.InputBindingInitRequest.newBuilder()
+        .setMetadata(ComponentProtos.MetadataRequest.newBuilder()
+            .putAllProperties(expectedMetadata)
+            .build())
+        .build());
+
+    assertThat(response).as("init return just a default InitResponse")
+        .isNotNull()
+        .isEqualTo(Bindings.InputBindingInitResponse.getDefaultInstance());
+    var capturedMetadata = ArgumentCaptor.forClass(Map.class);
+    verify(mockComponent).init(capturedMetadata.capture());
+    assertThat(capturedMetadata.getValue())
+        .as("An equivalent metadata map should be provided to the component")
+        .isNotNull()
+        .isEqualTo(expectedMetadata);
+  }
+
+  @Test
+  void initMonoErrorThrows() {
+    when(mockComponent.init(anyMap())).thenReturn(monoError());
+
+    assertThrowsGrpcRuntimeException(() -> client.init(Bindings.InputBindingInitRequest.getDefaultInstance()));
+  }
+
+
+  @Test
+  void pingHappyCase() {
+    when(mockComponent.ping()).thenReturn(Mono.empty());
```

Comment:
Does this assert that the (mocked) component's `ping()` method was, indeed, called?

Reply:
No, it does not. I am only setting up the mock but I am not verifying it was called -  I do it indirectly by checking the response. Let me add this verification.

I am actually trying to refactor all the ping/features/init tests into something that can be re-used to avoid the duplication.